packages/server/test/login.integration.ts

```diff
@@ -351,41 +355,155 @@ describe('Login Integration', () => {
+            .post('/khoury_login')
+            .send({
+              email: 'liu.i@northeastern.edu',
+              campus: 1,
+              first_name: 'Iris',
+              last_name: 'Liu',
+              photo_url: 'sdf',
+              courses: fallSecGroups,
+            })
+            .expect(201);
+
+          const newSectionGroups = await ProfSectionGroupsModel.findOne({
+            where: { profId: prof.id },
+          });
+          expect(newSectionGroups.sectionGroups).toEqual(fallSecGroups);
         });
 
-        expect(prof.courses).toHaveLength(0); // Does not create user courses
+        it('cycles between semesters', async () => {
+          const summerFull = await SemesterFactory.create({
+            season: 'Summer_Full',
+            year: 2022,
+          });
+          const summer2 = await SemesterFactory.create({
+            season: 'Summer_2',
+            year: 2022,
+          });
+          const course4 = await CourseFactory.create({ semester: summerFull });
+          const course5 = await CourseFactory.create({ semester: summer2 });
+
+          const sem = await SemesterModel.findOne({
+            where: { season: 'Fall', year: 2022 },
+          });
+          expect(sem).toBeUndefined();
+
+          await supertest()
+            .post('/khoury_login')
+            .send({
+              email: 'liu.i@northeastern.edu',
+              campus: 1,
+              first_name: 'Iris',
+              last_name: 'Liu',
+              photo_url: 'sdf',
+              courses: fallSecGroups,
+            })
+            .expect(201);
+
+          const newSem = await SemesterModel.findOne({
+            where: { season: 'Fall', year: 2022 },
+          });
+          expect(newSem).toBeDefined();
+
+          const oldCourses: CourseModel[] = [
+            course,
+            course2,
+            course3,
+            course4,
+            course5,
+          ];
+          for (const c of oldCourses) {
+            await c.reload();
+            expect(c.enabled).toBe(false);
+          }
+        });
 
-        const newSem = await SemesterModel.findOne({
-          where: { season: 'Fall', year: 2022 },
+        it('does not disable concurrent semester courses', async () => {
+          const summer1 = await SemesterFactory.create({
+            season: 'Summer_1',
+            year: 2022,
+          });
+          const course4 = await CourseFactory.create({ semester: summer1 });
+
+          await supertest()
+            .post('/khoury_login')
+            .send({
+              email: 'liu.i@northeastern.edu',
+              campus: 1,
+              first_name: 'Iris',
+              last_name: 'Liu',
+              photo_url: 'sdf',
+              courses: [
+                {
+                  crns: [23456],
+                  semester: '202250', // 2022 summer full
+                  name: 'Fundies 2 Accel',
+                },
+              ],
+            })
+            .expect(201);
+
```

Comment:
can we add an assert to expect the summer1 class to be disabled when a summer 2 course is added?